src/portals/hope/features/claims/hooks/use-claim.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  ClaimInformationMemberFragment,
+  ClaimInformationSubclaimFragment,
+  useClaimDetailsQuery,
+} from 'types/generated/graphql'
+import { ApolloError } from '@apollo/client'
+
+gql`
+  query ClaimDetails($claimId: ID!) {
+    claim(id: $claimId) {
+      id
+      state
+      registrationDate
+      transcriptions {
+        __typename
+      }
+      subclaims {
+        ...ClaimInformationSubclaim
+      }
+      member {
+        ...ClaimInformationMember
+      }
+    }
+  }
+
+  fragment ClaimInformationMember on Member {
+    memberId
+    firstName
+    lastName
+    identity {
+      firstName
+      lastName
+      nationalIdentification {
+        identification
+        nationality
+      }
+    }
+    sanctionStatus
+    contractMarketInfo {
+      market
+      preferredCurrency
+    }
+    directDebitStatus {
+      activated
+    }
+    payoutMethodStatus {
+      activated
+    }
+    adyenShopperReference
+    transactions {
+      id
+      amount {
+        amount
+        currency
+      }
+      timestamp
+      type
+      status
+    }
+  }
+
+  fragment ClaimInformationSubclaim on Subclaim {
+    id
+    type
+    createdAt
+    cause
+    reserve
+    propertySelections {
+      ...PropertySelection
+    }
+    contract {
+      id
+      preferredCurrency
+    }
+    trial {
+      id
+    }
+    agreement {
+      id
+      typeOfContract
+      carrier
+      termsAndConditions {
+        commencementDate
+        url
+        displayName
+      }
+    }
+    paymentOrders {
+      ...PaymentOrderInformation
+    }
+  }
+`
+
+const ClaimContext = createContext<{
+  claimId: string
+  claimState: string
+  claimRegistrationDate: string
+  claimTranscriptions: { __typename: 'ClaimTranscription' }[]
+  subclaims: ClaimInformationSubclaimFragment[]
+  getSubclaim: (
+    subclaimId: string,
+  ) => ClaimInformationSubclaimFragment | undefined
+  member: ClaimInformationMemberFragment
+  error: ApolloError | undefined
+}>({
+  claimId: '',
+  claimState: '',
+  claimRegistrationDate: '',
+  claimTranscriptions: [],
+  subclaims: [],
+  getSubclaim: () => ({} as ClaimInformationSubclaimFragment),
+  member: {
+    memberId: '',
+  },
+  error: undefined,
+})
+
+export const useClaim = () => useContext(ClaimContext)
+
+export const ClaimProvider: React.FC<{ claimId: string }> = ({
+  children,
+  claimId,
+}) => {
+  const { data, error } = useClaimDetailsQuery({ variables: { claimId } })
+  const claimState = data?.claim?.state ?? ''
+  const claimRegistrationDate = data?.claim?.registrationDate
+  const claimTranscriptions =
+    data?.claim?.transcriptions ??
+    ([] as { __typename: 'ClaimTranscription' }[])
+  const subclaims =
+    data?.claim?.subclaims ?? ([] as ClaimInformationSubclaimFragment[])
+  const member = data?.claim?.member ?? ({} as ClaimInformationMemberFragment)
+
+  const getSubclaim = (subclaimId: string) =>
+    subclaims.find((subclaim) => subclaim.id === subclaimId) ?? undefined
```

Comment:
Shouldn't `undefined` be the default?
```suggestion
    subclaims.find((subclaim) => subclaim.id === subclaimId)
```